src/nvim/api/vim.c

```diff
@@ -2615,3 +2616,29 @@ void nvim__put_attr(Integer id, Integer start_row, Integer start_col,
   decorations_add_luahl_attr(attr, (int)start_row, (colnr_T)start_col,
                              (int)end_row, (colnr_T)end_col);
 }
+
+/// Gets api function documentation
+//
+// @return Dictionary of the form
+// { "nvim_win_set_option": {...}, "nvim_win_get_option": {...}, ... }
+Dictionary nvim_doc_get_api(void)
```

Comment:
was hoping to instead introduce `nvim_get_api` which will deprecate `nvim_get_api_info`.

    nvim_get_api(Dict options)

works like the old `nvim_get_api_info`, except:

- `options.doc = true` (default=false) includes docstrings with the API metadata 
- the `functions` and `ui_events` elements are maps instead of a list
  - feel free to make this a TODO for this PR, if you don't want to do it